### PR TITLE
Websockets

### PR DESCRIPTION
**Pull request by ncalexan**
@rnewman, could you comment on the REST API for the UA service in https://github.com/mozilla/tofino/commit/f05d568467fc0b822db18dd2f67efa549e3bdc37.  I didn't spend a second thinking about it but together, we should :)

### CODE REVIEW

File: app/main/user-agent-service.js [hunk 1]
```diff
@@ -0,0 +1,220 @@
...
+
+    const errors = req.validationErrors();
+    if (errors) {
+      res.status(401).json(errors);
+      return;
+    }
+
+    const { q } = req.query;
+    const results = yield storage.visitedMatches(q);
+    res.json({ results });
```
Comment (rnewman) [inline app/main/user-agent-service.js]: We're probably going to want the richer types specified in `AwesomebarResult` in my current branch, or at least something more than URLs… unless we only want to support URL completion here.

File: app/main/browser.js [hunk 2]
```diff
@@ -231,48 +127,93 @@ app.on('window-all-closed', () => {
...
   }
 });
 
-ipc.on('instrument-event', (event, args) => {
-  // Until we transpile app/, we can't destructure in the argument list or inline here.
-  instrument.event(args.name, args.method, args.label, args.value);
-});
+async function newBrowserWindow(tabInfo: ?Object) {
+  const bw = await makeBrowserWindow(tabInfo);
+  browserWindowIds.push(bw.id);
```
Comment (rnewman) [inline app/main/browser.js]: Shouldn't this be returning something useful, like `bw`?

File: app/main/server.js [hunk 3]
```diff
@@ -0,0 +1,10 @@
+import * as service from './user-agent-service';
```
Comment (rnewman) [inline app/main/server.js]: License block.

File: app/shared/profile-diffs.js [hunk 4]
```diff
@@ -15,14 +15,14 @@
...
-export function bookmarks(bookmarkSet: [string]) {
+export function bookmarks(bookmarkSet: string[]) {
   return {
     type: profileDiffTypes.BOOKMARKS,
     payload: bookmarkSet,
   };
 }
 
-export function completions(text: string, completionList: [AwesomebarMatch]) {
+export function completions(text: string, completionList: AwesomebarMatch[]) {
```
Comment (rnewman) [inline app/shared/profile-diffs.js]: Thanks!

**Comment by rnewman** [comment]
I hate to say "LGTM" on something this substantial, but it does.

**Comment by bgrins** [comment]
I'm happy with the approach here

**Pull request merged**